dataclients/kubernetes/kube.go

```diff
@@ -467,11 +467,19 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 				if prule.Backend.Traffic > 0 {
 					r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
 					if err != nil {
-						// tolerate single rule errors
+						// fail when there's a temporary network error so that we keep the last known rules
+						// active until the next sync loop.
+						if err, ok := err.(net.Error); ok {
```

Comment:
note that checking for net.Error is unfortunately only reliable when we're directly handling the net package. Remember the io.EOF errors.

https://github.com/golang/go/issues/13667
and maybe more interestingly: https://github.com/golang/go/issues/16465#issuecomment-234669772

i think it would be better to do this:
- add an additional return argument to this function, which contains all the route ids that had an error, and just log the error without checking it just like before
- in the loadall(), keep the previous version of the failed routes
- in the loadupdate(), ignore the failed routes (make sure not to delete them)